src/pcm-iio.cpp

```diff
@@ -346,12 +346,37 @@ std::string build_csv_row(const std::vector<std::string>& chunks, const std::str
                            });
 }
 
+std::string get_root_port_dev(const bool show_root_port, int part_id,  const pcm::iio_stack *stack)
+{
+    char tmp[9] = "        ";
+    std::string rp_pci;
+
+    if (!show_root_port)
+        return rp_pci;
+
+    for (auto part = stack->parts.begin(); part != stack->parts.end(); part = std::next(part))
+    {
+        if (part->part_id == part_id)
+        {
+            std::snprintf(tmp, sizeof(tmp), "%02x:%02x.%02x", part->root_pci_dev.bdf.busno,
```

Comment:
Function should not be %02x, it should just be %x. I will fix it.

Reply:
ok, waiting for the update

Reply:
Done